tools/flaky-test-retryer/handler.go

```diff
@@ -57,72 +57,51 @@ func NewHandlerClient(githubAccount string) (*HandlerClient, error) {
 	}, nil
 }
 
-// expectedMsg checks that the message we received is one we want to process.
-func expectedMsg(msg *prowapi.ReportMessage) bool {
-	repos, err := getReportRepos()
-	if err != nil {
-		log.Printf("Failed getting reporter's repos: %v", err)
-		return false
-	}
-	expRepo := false
-	if len(msg.Refs) > 0 {
-		for _, repo := range repos {
-			if msg.Refs[0].Repo == repo {
-				expRepo = true
-				break
-			}
-		}
-	}
-	return expRepo && msg.Status == prowapi.FailureState && msg.JobType == prowapi.PresubmitJob
-}
-
 // Listen scans for incoming Pubsub messages, spawning a new goroutine for each
 // one that fits our criteria.
 func (hc *HandlerClient) Listen() {
 	log.Printf("Listening for failed jobs...\n")
 	for {
 		hc.pubsub.ReceiveMessageAckAll(hc, func(msg *prowapi.ReportMessage) {
-			if expectedMsg(msg) {
-				go hc.HandleMessage(msg)
-			} else {
-				log.Println("Job did not fit criteria - skipping")
+			data := NewJobData(msg)
+			if data.IsSupported() {
+				go hc.HandleJob(data)
 			}
 		})
 	}
 }
 
 // HandleMessage gets the job's failed tests and the current flaky tests,
 // compares them, and triggers a retest if all the failed tests are flaky.
-func (hc *HandlerClient) HandleMessage(msg *prowapi.ReportMessage) {
-	prefix := fmt.Sprintf("%s/pull/%d-%s", msg.Refs[0].Repo, msg.Refs[0].Pulls[0].Number, msg.JobName)
-	log.Printf("Job %s: fit all criteria - Starting analysis", prefix)
+func (hc *HandlerClient) HandleJob(jd *JobData) {
+	log.Printf("Job %s: fit all criteria - Starting analysis", jd.String())
 
-	failedTests, err := getFailedTests(msg.JobName, string(msg.JobType), msg.Refs[0].Repo, msg.Refs[0].Pulls[0].Number)
+	failedTests, err := jd.getFailedTests()
 	if err != nil {
-		log.Printf("Job %s: could not get failed tests: %v", prefix, err)
+		log.Printf("Job %s: could not get failed tests: %v", jd.String(), err)
```

Comment:
actually I meant a function inside `JobData` which wraps `log` or `log.Printf`, so you can do something like `jd.logf`